lib/tests/sources/lparstat_source.cc

```diff
@@ -7,20 +7,33 @@ using namespace whereami::sources;
 using namespace whereami::testing::lparstat;
 
 SCENARIO("Using the lparstat data source") {
-
-    WHEN("WPAR information is not available") {
-        lparstat_fixture lparstat_source {"5.3.0.0", "output/lparstat/lpar.txt"};
-        REQUIRE(lparstat_source.partition_number() == 16);
-        REQUIRE(lparstat_source.partition_name() == "aix-71-agent3");
-        REQUIRE(lparstat_source.wpar_key() == 0);
-        REQUIRE(lparstat_source.wpar_configured_id() == 0);
+    WHEN("Only LPAR information is available") {
+        THEN("LPAR fields are filled and WPAR fields are empty") {
+            lparstat_fixture lparstat_source {"5.3.0.0", "output/lparstat/lpar.txt"};
+            REQUIRE(lparstat_source.partition_number() == 16);
+            REQUIRE(lparstat_source.partition_name() == "aix-71-agent3");
+            REQUIRE(lparstat_source.wpar_key() == 0);
+            REQUIRE(lparstat_source.wpar_configured_id() == 0);
+        }
     }
 
     WHEN("WPAR information is available") {
```

Comment:
I'd probably change the structure to something like:

```
WHEN("LPAR information is available") {
  WHEN("WPAR information is unavailable") {
  }
  WHEN("WPAR information is available") {
  }
}
```

because you check the lpar information in both cases anyways.